src/OrbitQt/CaptureOptionsDialog.cpp

```diff
@@ -77,4 +79,36 @@ void CaptureOptionsDialog::ShowSourcePathsMappingEditor() {
   }
 }
 
+void CaptureOptionsDialog::SetCollectMemoryInfo(bool collect_memory_info) {
+  ui_->collectMemoryInfoCheckBox->setChecked(collect_memory_info);
+}
+
+bool CaptureOptionsDialog::GetCollectMemoryInfo() const {
+  return ui_->collectMemoryInfoCheckBox->isChecked();
+}
+
+void CaptureOptionsDialog::SetMemorySamplingPeriodNs(uint64_t memory_sampling_period_ns) {
+  ui_->memorySamplingPeriodNsLineEdit->setText(QString::number(memory_sampling_period_ns));
+}
+
+uint64_t CaptureOptionsDialog::GetMemorySamplingPeriodNs() const {
+  CHECK(!ui_->memorySamplingPeriodNsLineEdit->text().isEmpty());
+  bool valid = false;
+  uint64_t result = ui_->memorySamplingPeriodNsLineEdit->text().toULongLong(&valid);
+  CHECK(valid);
```

Comment:
Are we going to crash if the user complete this with a non-numeric value?

Reply:
The user is not able to input a non-numeric value here. We set `<set>Qt::ImhDigitsOnly</set>` in the ui, such that only digits are allowed.